deeppavlov/core/layers/bert.py

```diff
@@ -0,0 +1,233 @@
+        # tasks where we need a fixed dimensional representation of the segment.
+        # We "pool" the model by simply taking the hidden state corresponding to the first token. We assume that this
+        # has been pre-trained
+        self.pooler = tf.keras.layers.Dense(pooler_fc_size, activation='tanh', trainable=trainable, name='pooler/dense')
+
+    def call(self,
+             inputs: tf.Tensor,
+             training: Optional[bool] = None,
+             mask: Optional[bool] = None,
+             **kwargs) -> tf.Tensor:
+
+        emb = self.embed(inputs, training=training)
+        emb_norm_do = self.embed_dropout(self.embed_layer_norm(emb), training=training)
+
+        # always compute mask if it is not provided
+        if mask is None:
+            mask = tf.cast(tf.not_equal(inputs, self.pad_token_index), tf.int32)
+        enc = self.encoder(emb_norm_do, training=training, mask=mask)
+        # For classification tasks, the first vector (corresponding to [CLS]) is used as the "sentence vector". Note
+        # that this only makes sense because the entire model is fine-tuned.
+        po = self.pooler(tf.squeeze(enc[:, 0:1, :], axis=1))
+
+        if self.return_stack is None:
+            return po
+        elif self.return_stack:
+            raise NotImplementedError('Currently all encoder layers output could not be obtained. You can get '
+                                      'sequence output from the last encoder layer setting return_stack to False')
+        else:
+            return enc
+
+
+class TransformerBlock(tf.keras.layers.Layer):
+    """
+    One block of transformer architecture.
+
+    Args:
+        hidden_size: hidden size of the Transformer (d_model)
+        intermediate_size: the size of the intermediate dense layer
+        num_heads: number of attention heads in the Transformer
+        hidden_dropout_prob: dropout probability for the hidden layers
+        attention_probs_dropout_prob: dropout probability of the attention probabilities
+        intermediate_act_fn: the non-linear activation function to apply to the output of the intermediate dense layer
+        layer_norm_epsilon: some small number to avoid zero division during layer normalization
+        trainable: whether the layer's variables should be trainable
+        **kwargs: keyword arguments for base Layer class
+    """
+    def __init__(self,
+                 hidden_size: int = 768,
+                 intermediate_size: int = 3072,
+                 num_heads: int = 12,
+                 hidden_dropout_prob: float = 0.1,
+                 attention_probs_dropout_prob: float = 0.1,
+                 intermediate_act_fn: Union[str, callable] = gelu,
+                 layer_norm_epsilon: float = 1e-12,
+                 trainable: bool = True,
+                 **kwargs) -> None:
+        super().__init__(**kwargs)
+
+        self.supports_masking = True
```

Comment:
Is this a class variable? Should it be changeable after the object is created?

Reply:
It is a special Keras attribute used to determine whether to compute the output mask (call compute_mask method) while calling the layer instance. You probably can change it, but I don't know common scenarios when you need that.

Reply:
We should at least comment that thing about keras. Just a few words, maybe with link to keras docs.
Also I would recommend to move this parameter to class variables, because it is not object-specific.

Reply:
Ok, I will add some comments. This parameter should stay here because it is by default (from base layer class) set to False, I am just changing its value. 

Thanks for the review!
I will wait for some other reviewers, and then push all the changes including your suggestion.